userguide/tutorials/java_client.adoc

```diff
@@ -0,0 +1,63 @@
+ 
+== Writing the Java code
+
+Once the dependency is added, you can write code that invokes the Kill Bill APIs as shown below.
+ 
+[source,java]
+ int default_connection_timeout_sec = 10;
+ int default_read_timeout_sec = 60;
+ int default_request_timeout_sec = default_read_timeout_sec;
+ String username = "admin";
+ String password = "password";
+ String apiKey = "bob";
+ String apiSecret = "lazar";
+ String serverHost = "localhost";
+ int serverPort = 8080;
+ String kbServerUrl = String.format("http://%s:%d", serverHost, serverPort);
+ KillBillHttpClient killBillHttpClient = new KillBillHttpClient(kbServerUrl, username, password, apiKey,apiSecret, null, null, default_connection_timeout_sec * 1000, default_read_timeout_sec * 1000,default_request_timeout_sec * 1000);
+ AccountApi accountApi = new AccountApi(killBillHttpClient);
+ UUID accountId = UUID.fromString("a21f1ca3-53ec-456b-8039-7170350c9c12");
+ String createdBy = "Kill Bill Client Tutorial";
+ String reason = "Demonstrating Kill Bill Client";
+ String comment = "Demonstrating Kill Bill Client";
+ RequestOptions requestOptions =  RequestOptions.builder().withCreatedBy(createdBy).withReason(reason).withComment(comment).build();
+ ImmutableMap<String, String> NULL_PLUGIN_PROPERTIES = null;
+ List<PaymentMethod> paymentMethods = accountApi.getPaymentMethodsForAccount(accountId, NULL_PLUGIN_PROPERTIES,requestOptions);
+ logger.info("Payment methods=" + paymentMethods.size());
+
+
+ * This code first declares and initializes some variables like `username`, `password` and so on. 
+ 
+ * Next, a https://github.com/killbill/killbill-client-java/blob/9634a6d114ab71c868e7ef9ddc8a987cfec414ab/src/main/java/org/killbill/billing/client/KillBillHttpClient.java[KillBillHttpClient] instance is created using the variables declared earlier.  
+ 
+ * After that, the code creates an https://github.com/killbill/killbill-client-java/tree/9634a6d114ab71c868e7ef9ddc8a987cfec414ab/src/main/java/org/killbill/billing/client/api/gen[AccountApi] instance using the `killBillHttpClient` object. `AccountApi` can be used to execute account related Kill Bill methods. Just like `AccountApi`, the https://github.com/killbill/killbill-client-java/tree/9634a6d114ab71c868e7ef9ddc8a987cfec414ab/src/main/java/org/killbill/billing/client/api/gen[org.killbill.billing.client.api.gen] package has other classes  like `PaymentMethodApi`, `OverDueApi` which can be used to invoke the corresponding API methods.
+ 
+ * Next, the code declares and initializes some more variables like `createdBy`, `reason` and `comment`.
+ 
+ * A https://github.com/killbill/killbill-client-java/blob/9634a6d114ab71c868e7ef9ddc8a987cfec414ab/src/main/java/org/killbill/billing/client/RequestOptions.java[RequestOptions] instance is then created using these variables.
+
+* Finally, the code invokes the `AccountApi#getPaymentMethodsForAccount` method. This returns the payment methods corresponding to the specified `accountId`.
+
+== Executing the code
```

Comment:
Could you add a section on how to debug errors?

See https://groups.google.com/g/killbilling-users/c/CwiswTmGwTg/m/X5RB5h0RCAAJ for an example, specifically about catching `KillBillClientException` and how to find logs on the Kill Bill side (this could also just link to the other guide that goes through this in more details).